src/index.ts

```diff
@@ -371,6 +377,178 @@ export const createWapeSaleInstance = (
   return instance;
 };
 
+export const createGenSaleInstance = (
```

Comment:
I know in the past we had all the `create<name>SaleInstance` functions here, but it makes this file really bloated and hard to navigate IMO. It would be really nice if you could separate these into their logical groups in the `actions` folders. Maybe put each exported const in the `index.ts` of each index.ts for the actions it belongs to, then have one `createSaleInstance` here that calls whichever set of actions we actually need.

Reply:
Is this something worth focusing on in the gensale? Or should we accomplish this in the generic sale contract?

Reply:
The generic sale contract is generic by definition and won't have all the `create<name>SaleInstance` variations that this does, but it can be left as it was not in scope for this story. This is more a cleanliness thing that can be addressed in future work

Reply:
this is related to my previous comment about copying and rewriting the same code.
both of you are right here. what I would say - let's think about structuring code before we write it and copy.
for this one, we can keep it as is. just remember, someone will have to come here later to deal with all this, instead of working on some interesting task :)